File: pset6/mario.py
import cs50

def main():
    
    while True:
        print("Enter the height of the desired pyramid: ", end="")
        height = cs50.get_int()
        
        if height < 1 or height > 23:
            print("Invalid input! Please enter an interger between 1 and 23.")
            
        if height > 0 and height <= 23:
            break
    
    PrintPyramid(height)

    
def PrintPyramid(height):
    for row in range(height):
        PrintSpaces(height, row)
        PrintHashes(row)
        print("  ", end="")
        PrintHashes(row)
        print("")
        
def PrintSpaces(height, row):
    for column in range(height - row - 1):
        print(" ", end="")
    return
    
def PrintHashes(row):
    for column in range(row + 2):
        print("#", end="")
    return

if __name__ == "__main__":
    main()
    

File: pset1/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int height;
    do
    {
        printf("Enter the height of the desired pyramid:\n");
        height = get_int();
        if (height <= 0 || height >= 23)
        {
            printf("Invalid input! Please enter an interger between 1 and 23.\n");
        }
    }
    while (height <= 0 || height >= 23);

    for (int rows = 0; rows < height; rows++)
    {
        for (int column = 0; column < height - rows - 1; column++)
        {
            printf(" ");
        }
        for (int column2 = 0; column2 < rows + 2; column2++)
        {
            printf("#");
        }
        for (int column3 = 0; column3 < 2; column3++)
        {
            printf(" ");
        }
        for (int column4 = 0; column4 < rows + 2; column4++)
        {
            printf("#");
        }
        printf("\n");
    }
    
}
